atlas/AtlasMl/atlasml/clients/weaviate.py

```
@@ -1,39 +1,107 @@
+        try:
+            self.client = weaviate.connect_to_local(
+                host=weaviate_settings.host,
+                port=weaviate_settings.port,
+                grpc_port=weaviate_settings.grpc_port,
+            )
+            logger.info(
+                f"✅ Connected to Weaviate at {weaviate_settings.host}:{weaviate_settings.port}"
+            )
+        except WeaviateConnectionError as e:
+            logger.error(f"❌ Failed to connect to Weaviate: {e}")
+            raise WeaviateConnectionError(f"Could not connect to Weaviate server: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)